crates/core/tedge/src/cli/mqtt/subscribe.rs

```diff
@@ -158,6 +140,18 @@ fn subscribe(cmd: &MqttSubscribeCommand) -> Result<(), anyhow::Error> {
         }
     }
 
+    let _ = client.disconnect();
+    for event in connection.iter() {
+        match event {
+            Err(_)
+            | Ok(Event::Incoming(Incoming::Disconnect))
+            | Ok(Event::Outgoing(Outgoing::Disconnect)) => {
```

Comment:
Is this `Outgoing::Disconnect` correct? If we break before the other server acknowledges we're disconnected, that still feels incorrect. Maybe we need to `return` instead of `break` where we receive `Incoming::Disconnect` above (in the main loop, not this loop), since I don't think we want to disconnect twice?

Reply:
> Is this `Outgoing::Disconnect` correct? If we break before the other server acknowledges we're disconnected, that still feels incorrect.

It is correct. I checked the disconnection on the broker.

An `Incoming::Disconnect` is only received when the disconnect is initiated by the broker.  The only event that could be used as a sentinel for an actual disconnect is `Err(MqttState(ConnectionAborted))`

> Maybe we need to `return` instead of `break` where we receive `Incoming::Disconnect` above (in the main loop, not this loop), since I don't think we want to disconnect twice?

Disconnecting twice is harmless.